custom/enikshay/management/commands/mark_successfull_99dots_records.py

```diff
@@ -0,0 +1,110 @@
+
+        cases_to_update = set()
+        print("Filtering successful cases")
+        for repeat_record in with_progress_bar(records, length=count):
+            if repeat_record.attempts and any(
+                    (existing_message in attempt.message if attempt.message is not None else "")
+                    for attempt in repeat_record.attempts):
+                episode = accessor.get_case(repeat_record.payload_id)
+                if episode.get_case_property('dots_99_registered') != 'true':
+                    cases_to_update.add(episode)
+
+        with open('set_99dots_to_registered.csv', 'w') as f:
+            writer = csv.writer(f)
+            writer.writerow([
+                'beneficiary_id',
+                'episode_id',
+                'UpdatePatient Status',
+                'Adherence Status',
+                'TreatmentOutcome Status'
+            ])
+            print("Updating {} successful cases in 99DOTS".format(len(cases_to_update)))
+            for case in with_progress_bar(cases_to_update):
+                writer.writerow([
+                    get_person_case(domain, case.case_id).case_id,
+                    case.case_id,
+                    self.update_registered_status(domain, case),
+                    self.update_patients(domain, case),
+                    self.send_adherence(domain, case),
+                    self.send_treatment_outcome(domain, case),
+                ])
+
+    def update_registered_status(self, domain, case):
+        if self.commit:
+            update_case(domain, case.case_id, {'dots_99_registered': 'true'})
+
+    def update_patients(self, domain, case):
+        repeater_id = 'b4e19fd859f852871703e8e32a1764a9'
+        repeater_type = 'NinetyNineDotsUpdatePatientRepeater'
+        return self.send_repeat_record(domain, case.case_id, repeater_id, repeater_type)
+
+    def send_adherence(self, domain, case):
+        repeater_id = '9cf9a0e2df49b6271573afe677725dc2'
+        repeater_type = 'NinetyNineDotsAdherenceRepeater'
+
+        adherence_cases = get_adherence_cases_from_episode(domain, case.case_id)
+        if adherence_cases:
+            return {
+                adherence_case.case_id: self.send_repeat_record(domain, case.case_id, repeater_id, repeater_type).state
+                for adherence_case in adherence_cases
+            }
+        else:
+            return "No Adherence Cases"
+
+    def send_treatment_outcome(self, domain, case):
+        repeater_id = '3d912c5d00c73e9de5eb6a11c52a7301'
+        repeater_type = 'NinetyNineDotsTreatmentOutcomeRepeater'
+        if case.get_case_property('treatment_outcome', '') != '':
+            return self.send_repeat_record(domain, case.case_id, repeater_id, repeater_type).state
+        else:
+            return "No Treatment Outcome"
+
+    def send_repeat_record(self, domain, case_id, repeater_id, repeater_type):
+        repeat_record = RepeatRecord(
```

Comment:
Why create a new repeat record rather than a new attempt for the current one?

Reply:
hmm.. nice catch.
I think probably because the records are already `SUCCESS` ones or to track these retriggers separately?

Reply:
I think in most cases we don't actually have a repeat record to trigger from (these cases were never "successful" for the first repeater, which bubbles down to mean the other repeaters were never triggered at all). I could check if any of these exist first, I suppose.

Reply:
Makes sense - so the top-level one has a record, but the others likely don't?  In that case, I'd only bother special-casing the top-level one if you think it'll help your auditability down the line.  This seems reasonable as-is.

Reply:
The top level one isn't resent at all, we just check that it exists.